microsoft/testsuites/core/timesync.py

```diff
@@ -64,3 +74,78 @@ def timesync_validate_ptp(self, environment: Environment, node: Node) -> None:
+        """,
+        priority=1,
+    )
+    def timesync_check_clocksource(self, node: Node) -> None:
+        # 1. Check clock source name is one of hyperv_clocksource_tsc_page,
+        #  lis_hv_clocksource_tsc_page, hyperv_clocksource.
+        clocksource = [
+            "hyperv_clocksource_tsc_page",
+            "lis_hyperv_clocksource_tsc_page",
+            "hyperv_clocksource",
+        ]
+        cat = node.tools[Cat]
+        clock_source_result = cat.run(self.current_clocksource)
+        assert_that([clock_source_result.stdout]).described_as(
+            f"Expected clocksource name is one of {clocksource},"
+            f" but actual it is {clock_source_result.stdout}."
+        ).is_subset_of(clocksource)
+
+        # 2. Check CPU flag contains constant_tsc from /proc/cpuinfo.
+        lscpu = node.tools[Lscpu]
+        if "x86_64" == lscpu.get_architecture():
+            cpu_info_result = cat.run("/proc/cpuinfo")
+            assert_that(cpu_info_result.stdout).described_as(
+                "Expected 'constant_tsc' shown up in cpu flags."
+            ).contains("constant_tsc")
+
+        # 3. Check clocksource name shown up in dmesg.
+        dmesg = node.tools[Dmesg]
+        assert_that(dmesg.get_output()).described_as(
+            f"Expected clocksource {clock_source_result.stdout} shown up in dmesg."
+        ).contains(f"clocksource {clock_source_result.stdout}")
+
+        # 4. Unbind current clock source if there are 2+ clock sources,
+        # check current clock source can be switched to a different one.
+        if node.shell.exists(PurePosixPath(f"{self.unbind_clocksource}")):
+            available_clocksources = cat.run(self.available_clocksource)
+            available_clocksources_array = available_clocksources.stdout.split(" ")
+            # We can not unbind clock source if there is only one existed.
+            if len(available_clocksources_array) >= 2:
+                available_clocksources_array.remove(clock_source_result.stdout)
+                cmd_result = node.execute(
+                    f"sh -c 'echo {clock_source_result.stdout} > "
+                    f"{self.unbind_clocksource}'",
+                    sudo=True,
+                )
+                assert_that(cmd_result.exit_code).described_as(
+                    f"Fail to execute command "
+                    f"[echo {clock_source_result.stdout} > {self.unbind_clocksource}]."
+                ).is_equal_to(0)
+                timout_timer = create_timer()
+                timeout = 30
+                while timout_timer.elapsed(False) < timeout:
+                    current_clock_source_result = cat.run(
+                        self.current_clocksource, force_run=True
+                    )
+                    if (
+                        current_clock_source_result.stdout
+                        in available_clocksources_array
+                    ):
+                        break
```

Comment:
Add a sleep here to prevent access too fast locally, and slow down other things. May be 0.5 seconds.